controllers/pipelines/run_controller_decoupled_test.go

```diff
@@ -34,6 +34,10 @@ var _ = Describe("Run controller k8s integration", Serial, func() {
 				g.Expect(run.Status.ProviderId.Provider).To(Equal(testConfig.DefaultProvider))
 			})).Should(Succeed())
 
+			Expect(runHelper.Update(func(run *pipelinesv1.Run) {
+				run.Spec = pipelinesv1.RandomRunSpec()
+			})).NotTo(Succeed())
```

Comment:
Is it possible to verify that it's the expected error and not failing for a different reason?